app/lib/email.server.ts

```diff
@@ -89,6 +91,40 @@ function maybeThrowSES(e: any, warning: string) {
   ])
 }
 
+/**
+ * Amazon SES has a byte limit of 262144 for TemplateData arguments.
+ * If the circular body is over this size limit, the message will be truncated.
+ * If the body is truncated, the circular website link will be removed.
+ * A warning that the email has been truncated is added and the circular website link is added back in.
+ * An additional buffer is allowed for for variation in url length and truncation warning.
+ */
+export function truncateEmailBodyIfNecessary({
+  body,
+  circularId,
+  origin,
+}: {
+  body: string
+  circularId?: number
+  origin?: string
+}) {
+  const textBody = getBody(body)
+  const inputBytes = Buffer.byteLength(textBody, 'utf-8')
+
+  if (inputBytes <= 262144) return textBody
+
+  const buffer = Buffer.from(textBody, 'utf-8')
+  const truncatedBuffer = buffer.subarray(0, 261900)
+  const truncatedBody = truncatedBuffer.toString('utf-8')
```

Comment:
This code encodes the text to bytes as UTF-8, truncates, then decodes as UTF-8. This is unsafe because UTF-8 is a variable length encoding. I suggest searching for a JavaScript library to summarize or truncate text that safely handles variable-length encodings.